server/routes/users/index.ts

```diff
@@ -20,6 +48,15 @@ app.get('/api/v1/token', validateRefreshJWT, wrap(userController.getAccessToken)
 app.delete('/api/v1/logout', validateRefreshJWT, wrap(userController.logout));
 
 // get user reactions
-app.get('/api/v1/users/reactions', validateRefreshJWT, wrap(userController.getUserReactions));
+app.get('/api/v1/users/:username/reactions', validateUsernameParam, wrap(userController.getUserReactions));
+
+// get user data such as reputation and number of posts
+app.get('/api/v1/users/:username', validateUsernameParam, wrap(userController.getUserData));
+
+// change the user image
+app.post('/api/v1/users/change-icon', validateRefreshJWT, validateUsernameCookie, upload.single('image'), wrap(userController.changeUserIcon));
```

Comment:
This is the beauty of REST. You can and should use the same route for GET and POST of the same resource.

And this will help you to be consistent with validation as well.

```suggestion
app.post('/api/v1/users/:username/icon', validateRefreshJWT, validateUsernameCookie, upload.single('image'), wrap(userController.changeUserIcon));
```